### PR TITLE
deps: pass Windows SDK version to ChakraCore

### PR DESCRIPTION
**Pull request by joaocgreis**
This ensures ChakraCore is built with the same Windows SDK as the rest of the project.

CI: https://ci.nodejs.org/view/All/job/chakracore-test/77/

cc @digitalinfinity @kfarnung 

##### Checklist

- [x] `make -j4 test` (UNIX), or `vcbuild test` (Windows) passes
- [x] commit message follows [commit guidelines](https://github.com/nodejs/node/blob/master/CONTRIBUTING.md#commit-message-guidelines)

##### Affected core subsystem(s)

deps, build, chakracore

### CODE REVIEW

**Review by jianchun — APPROVED** [review]
LGTM

**Review by kfarnung — APPROVED** [review]
🚢

File: deps/chakrashim/chakracore.gyp [hunk 1]
```diff
@@ -110,6 +110,7 @@
                 '/p:Configuration=$(ConfigurationName)',
                 '/p:RuntimeLib=<(component)',
                 '/p:AdditionalPreprocessorDefinitions=COMPILE_DISABLE_Simdjs=1',
+                '/p:WindowsTargetPlatformVersion=$(WindowsTargetPlatformVersion)',
```
Comment (kunalspathak) [inline deps/chakrashim/chakracore.gyp:113]: isn't this only require for ARM? See [this](https://github.com/nodejs/node-chakracore/commit/262f7e0ac649c4b1595808fd7eea688c78b6b7ee#diff-980723060c448bbe13349f168d8a0bdb)

Comment (kunalspathak) [reply deps/chakrashim/chakracore.gyp:113]: I think this is needed in general for VS2017. Just happened to read [this](https://github.com/Microsoft/ChakraCore/issues/2348#issuecomment-304963588)

**Comment by digitalinfinity** [comment]
@curtisman is making a change to clean this up in Microsoft/ChakraCore- lets hold this change till that PR lands

**Pull request closed without merging**